src/components/Button/styles.module.css

```diff
@@ -13,75 +23,242 @@
+    var(--OLED, var(--white-t-50));
+  &.bg-lighten {
+    &:hover {
+      background-color: var(--lighten-100);
+      border: 1px solid var(--LIGHT, transparent) var(--DARK, var(--white-t-60))
+        var(--OLED, var(--white-t-50));
+    }
+    &:active {
+      background-color: var(--lighten-200);
+      border: 1px solid var(--LIGHT, transparent) var(--DARK, var(--white-t-50))
+        var(--OLED, var(--white-t-50));
+    }
+  }
+  &.bg-darken {
+    &:hover {
+      background-color: var(--darken-100);
+      border: 1px solid var(--LIGHT, transparent) var(--DARK, var(--white-t-60))
+        var(--OLED, var(--white-t-50));
+    }
+    &:active {
+      background-color: var(--darken-200);
+      border: 1px solid var(--LIGHT, transparent) var(--DARK, var(--white-t-50))
+        var(--OLED, var(--white-t-50));
+    }
   }
 }
 
 .outline {
-  background: var(--button-outline-background);
   color: var(--button-outline-text-color);
+  background-color: var(--default-outline-background);
   border: var(--button-outline-border);
+  &.bg-lighten {
+    &:hover {
+      color: var(--lighten-100);
+      background-color: var(--outline-lighten-bg);
+      border: 1px solid var(--lighten-100);
+    }
+    &:active {
+      color: var(--lighten-200);
+      background-color: var(--outline-lighten-bg);
+      border: 1px solid var(--lighten-200);
+    }
+  }
+  &.bg-darken {
+    &:hover {
+      /*In dark mode: against dark bg, we add logic to mtaintain text to higher brighteness.*/
+      color: var(--LIGHT, var(--darken-100)) var(--DARK, var(--lighten-100))
+        var(--OLED, var(--lighten-100));
+      background-color:  var(--LIGHT, transaprent) var(--DARK, var(--translucent-5))
+        var(--OLED, var(--translucent-5));
+      border: 1px solid var(--darken-100);
+    }
+    &:active {
+      color: var(--LIGHT, var(--darken-200)) var(--DARK, var(--lighten-200))
+        var(--OLED, var(--lighten-200));
+      background-color:  var(--LIGHT, transaprent) var(--DARK, var(--translucent-25))
+        var(--OLED, var(--translucent-25));
+      border: 1px solid var(--darken-200);
+    }
+  }
+}
+
+.borderless {
+  color: var(--defaultColor);
+  background-color: transparent;
+  border: var(--border);
+  &.bg-lighten {
+    &:hover {
+      color: var(--lighten-100);
+      background-color: transparent;
+    }
+    &:active {
+      color: var(--lighten-200);
+      background-color: transparent;
+    }
+  }
 
-  &:hover,
-  &:focus {
-    background: var(--button-outline-hover-background);
-    color: var(--button-outline-hover-text-color);
-    border: var(--button-outline-hover-border);
+  &.bg-darken {
+    &:hover {
+      color: var(--LIGHT, var(--darken-100)) var(--DARK, var(--lighten-100))
+        var(--OLED, var(--lighten-100));
+      background-color:  var(--LIGHT, var(--translucent-5)) var(--DARK, var(--translucent-5))
+        var(--OLED, var(--translucent-25));
```

Comment:
Is there some way we can eliminate this conditional logic from the button stylesheet? Ideally components should not have to think about themes, they should just reference logical variables (not color variables) from the theme stylesheet, to keep the theming all in one place

If you remove the `bg-darken` and `bg-lighten` that might make it easier 🤔

Reply:
Yep, removing bg-darken/lighten property will make it easier to rid of those conditional logic, tho they can be moved to global stylesheet as well. But i will follow thru with the first review, remove them.